### ISSUE TITLE
Blocked element reappear after page reload when block rule is still valid

### ISSUE THREAD

**Issue by GeetaSarvadnya** [brave/brave-browser#19229]
## Description 

Blocked element reappeared after page reload when an element is still blocked in brave:/adblock  
## Steps to Reproduce

   1. Clean profile 1.33.61
   2. Open brave.com
   3. Block the `Download Brave` button on the home page
   4. Confirm blocked element is added in brave://adblock 
   5. Reload brave.com, `Download Brave` button reappeared on the webpage

## Actual result:

`Download Brave` button reappeared when brave.com is refreshed 

## Expected result:
`Download Brave` button should be blocked on the webpage until the blocked element is removed from brave://adblock
## Reproduces how often: 

Easy

## Brave version (brave://version info)

<html>
<body>

Brave | 1.33.61 Chromium: 96.0.4664.27 (Official Build) nightly (64-bit)
-- | --
Revision | 0658dae02ef18b530eda4384920ee23f5e156c55-refs/branch-heads/4664@{#459}
OS | Windows 10 Version 21H1 (Build 19043.1288)

</body>
</html>

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes in 1.31.88, But not in 1.30.86
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? NA
- Does the issue resolve itself when disabling Brave Rewards? NA
- Is the issue reproducible on the latest version of Chrome? NA

## Miscellaneous Information:

cc: @brave/legacy_qa @mkarolin @rebron

**Comment by ryanbr** [comment]
What blocked element are you using @GeetaSarvadnya  ? try adding this: `brave.com##.site-header` into `brave://adblock`  which will hide the top bar on brave.com. See if you can replicate it with that filter

**Comment by GeetaSarvadnya** [comment]
@ryanbr 
![adblock_CR96](https://user-images.githubusercontent.com/38657976/140374595-2b855f98-7dc0-46f0-86d2-7a2db38fe76f.gif)

**Comment by ryanbr** [comment]
Switch to Aggresive mode in shields @GeetaSarvadnya Also `brave.com##.download` would be the correct filter here. I would've thought aggressive mode wouldn't be needed for cusom filters cc: @antonok-edm

**Comment by antonok-edm** [comment]
@ryanbr I did find https://github.com/brave/brave-browser/issues/19085, which also only applies to standard mode but it shouldn't be affecting rules with a hostname attached - will investigate

**Comment by rebron** [comment]
+ 1 from here https://github.com/brave/brave-browser/issues/19260 with nice str/use case.

**Comment by antonok-edm** [comment]
Two things going on here:
1. `brave.com` is being considered a "vetted search engine page", where cosmetic filtering is disabled.
2. _all_ cosmetic filtering is disabled on these vetted pages including user-created cosmetic filters; rather than just filters from the default lists.

I believe `1` was previously acknowledged and considered acceptable, but `2` is a regression and should be fixed.

**Comment by tortino** [comment]
[#19260](https://github.com/brave/brave-browser/issues/19260) was closed as a duplicate of this bug, but it was not related to vetted pages: I reported about arstechnica.com.
Does this fix also applies to #19260?

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.36.47 Chromium: 97.0.4692.71 (Official Build) nightly (x86_64)
-- | --
Revision | adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}
OS | macOS Version 11.6.1 (Build 20G224)

### Steps:
1. new profile
2. loaded `brave.com`
3. right-clicked on the `Download Brave` button and chose `Brave` -> `Block element`
4. selected the download-button region and clicked on `Create`
5. confirmed the `Download Brave` button disappeared
6. checked via `brave://adblock` and confirmed `brave.com###home-download-button-hero` was added to the block-rule list

Also did the same, but for the gray space at the top of `arstechnica.com` (from https://github.com/brave/brave-browser/issues/19260).

### brave.com

example | example | example | example
---------|----------|---------|---------
<img width="1312" alt="Screen Shot 2022-01-19 at 15 00 19" src="https://user-images.githubusercontent.com/387249/150232772-e4eb85b1-0fa2-4b88-8481-088a24c1a9f9.png"> | <img width="1312" alt="Screen Shot 2022-01-19 at 15 00 43" src="https://user-images.githubusercontent.com/387249/150232810-4981d5e6-5fd6-4f14-8397-6ebb43664f47.png"> | <img width="1312" alt="Screen Shot 2022-01-19 at 15 00 57" src="https://user-images.githubusercontent.com/387249/150232850-1f301d74-c276-450d-a395-9a59a3a189f4.png"> | <img width="1312" alt="Screen Shot 2022-01-19 at 15 01 09" src="https://user-images.githubusercontent.com/387249/150232886-d3cb48bc-f2f3-469a-96e5-f87244694594.png">





### arstechnica.com

example | example | example | example
---------|----------|---------|---------
<img width="1312" alt="Screen Shot 2022-01-19 at 15 02 01" src="https://user-images.githubusercontent.com/387249/150232643-670298cc-88d3-4568-9e42-83412cc519c9.png"> | <img width="1312" alt="Screen Shot 2022-01-19 at 15 02 10" src="https://user-images.githubusercontent.com/387249/150232655-fed3a0c3-ad02-47ff-951f-9bf2c7335ecc.png"> | <img width="1312" alt="Screen Shot 2022-01-19 at 15 02 22" src="https://user-images.githubusercontent.com/387249/150232660-bc267e75-a8a5-416c-8405-63b2a90c54e8.png"> | <img width="1312" alt="Screen Shot 2022-01-19 at 15 02 15" src="https://user-images.githubusercontent.com/387249/150232705-b7abbfe6-021a-4433-b147-71a0266c77ea.png">

**Comment by antonok-edm** [comment]
@tortino I think #19260 is better described as a duplicate of https://github.com/brave/brave-browser/issues/20177 (also fixed) - I'll update that issue as well, thanks!
